### PR TITLE
docs: Add pre-commit hook with 🚫💩 lint-staged section to the README

### PR DESCRIPTION
**Pull request by okonet**
As discussed on twitter: https://twitter.com/dan_abramov/status/831616492068290561

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,6 +124,34 @@ prettier --write 'src/**/*.js' 'bin/*.js'
...
+{
+	"scripts": {
+		"lint:staged": "lint-staged"
+	},
+  "lint-staged": {
+	  "*.js": [
+		  "prettier --write",
+		  "git add"
+	  ]
+  }
```
Comment (sapegin) [inline README.md]: ,

Comment (sapegin) [inline README.md]: And also mixed tabs and spaces.

Comment (okonet) [inline README.md]: Fixed both.

File: README.md [hunk 2]
```diff
@@ -124,6 +124,34 @@ prettier --write 'src/**/*.js' 'bin/*.js'
...
+
+```json
+{
+  "scripts": {
+    "lint:staged": "lint-staged"
+  },
+  "lint-staged": {
+    "*.js": [
+      "prettier --write",
+      "git add"
```
Comment (jlongster) [inline README.md]: You need to "re-invoke" git add?

**Comment by jlongster** [comment]
We really should create better docs and split them up more, probably inside `docs`. I'm fine merging as is though but I'll try to find time soon to improve the docs.

File: README.md (hunk 2 above)
Comment (okonet) [inline README.md]: Yeah, otherwise those files won't be added to the same commit.

**Comment by vjeux** [comment]
Thanks for adding this!

File: README.md (hunk 2 above)
Comment (smably) [inline README.md]: Won't this also add any changes in the working dir that were not yet staged? So if I partially stage some files (or stage them and then fix another issue before committing), prettier will run on the version in the working dir and then stage that?

Should probably `git stash` before running prettier, and `git stash pop` after `git add`, no?

Comment (okonet) [inline README.md]: No, it won't. Please read the 🚫💩 lint-staged docs. Shortly, it calls those commands with staged files as arguments.

Comment (smably) [inline README.md]: Ah, I read the README but missed that section. In that case, looks good. 👍

**Comment by jlongster** [comment]
Let's do it. Thanks a lot!

**Comment by wangzuo** [comment]
Why its `"lint:staged": "lint-staged"` instead of `"lint-staged": "lint-staged"` according to [lint-staged](https://github.com/okonet/lint-staged#installation--setup) ?

**Comment by okonet** [comment]
@wangzuo it doesn't matter. Use whatever you prefer.

**Comment by wangzuo** [comment]
@okonet the current config on master is not consistent.

**Comment by sapegin** [comment]
@wangzuo Not consistent with what? ;-)

**Comment by wangzuo** [comment]
It seems the config is right here, but somehow changed later.

**Comment by okonet** [comment]
I don't think I'm following. It doesn't work with this config for you or what's the problem?

**Comment by wangzuo** [comment]
@okonet I guess https://github.com/prettier/prettier/commit/c8c91d4af3bc2e58427583c463afe9dd2a357fd4 is unnecessary.

**Pull request merged**